src/utils/proxy.js

```diff
@@ -106,9 +129,9 @@ const serveRedirect = async function ({ req, res, proxy, match, options }) {
   if (isFunction(options.functionsPort, req.url)) {
     return proxy.web(req, res, { target: options.functionsServer })
   }
-  const urlForAddons = getAddonUrl(options.addonUrls, req)
+  const urlForAddons = getAddonUrl(options.addonsUrls, req)
   if (urlForAddons) {
-    return proxy.web(req, res, { target: urlForAddons })
```

Comment:
This doesn't work since you need to change the request host to match the certificate of the target.
I believe the intention of this code was to redirect `http://localhost:8888/.netlify/<addon>/*` to `https://<site-url>/.netlify/<addon>/*`